pkg/apis/core/validation/validation.go

```diff
@@ -1187,7 +1187,7 @@ func validateHostPathType(hostPathType *core.HostPathType, fldPath *field.Path)
 	allErrs := field.ErrorList{}
 
 	if hostPathType != nil && !supportedHostPathTypes.Has(string(*hostPathType)) {
-		allErrs = append(allErrs, field.NotSupported(fldPath, hostPathType, supportedHostPathTypes.List()))
+		allErrs = append(allErrs, field.NotSupported(fldPath, hostPathType, sets.List[string](supportedHostPathTypes)))
```

Comment:
If we keep it as string, you still don't need `[string]` - it's inferred by the argument